lib/analytics.js

```diff
@@ -678,22 +694,36 @@ Analytics.prototype._invoke = function(method, facade) {
 
   var failedInitializations = self.failedInitializations || [];
   each(function(integration, name) {
-    if (!facade.enabled(name)) return;
+    var facadeCopy = extend(true, {}, facade);
+
+    if (!facadeCopy.enabled(name)) return;
     // Check if an integration failed to initialize.
     // If so, do not process the message as the integration is in an unstable state.
     if (failedInitializations.indexOf(name) >= 0) {
       self.log(
-        'Skipping invokation of .%s method of %s integration. Integation failed to initialize properly.',
+        'Skipping invocation of .%s method of %s integration. Integration failed to initialize properly.',
         method,
         name
       );
     } else {
       try {
-        metrics.increment('analytics_js.integration.invoke', {
-          method: method,
-          integration_name: integration.name
+        // Apply any integration middlewares that exist, then invoke the integration with the result.
+        self._integrationMiddlewares.applyMiddlewares(facadeCopy.obj, function(
+          payload
+        ) {
+          // A nullified payload should not be sent to an integration.
+          if (payload === null) {
+            self.log('Payload to integration "%s" was null and dropped.', name);
+            return;
+          }
+          facadeCopy.obj = payload;
+          metrics.increment('analytics_js.integration.invoke', {
+            method: method,
+            integration_name: integration.name
+          });
+
+          integration.invoke.call(integration, method, facadeCopy);
```

Comment:
Any reason to do this vs `integration.invoke(method, facadeCopy)`?

Reply:
In this case, I'm just copying the function call exactly as it was before. Avoiding playing with fire of this old code. 😆